File: Data Structures & Algorithms/subsets/submission-0.java
class Solution {
    List<Integer> subset = new ArrayList<>();
    Set<List<Integer>> powerSet = new HashSet<>();

    private void backtrack(int[] nums, int idx){
        if(idx == nums.length){
            powerSet.add(new ArrayList<>(subset));
            return;
        }

        subset.add(nums[idx]);
        backtrack(nums, idx + 1);
        subset.remove(subset.size() - 1);
        backtrack(nums, idx + 1);
    }

    public List<List<Integer>> subsets(int[] nums) {
        backtrack(nums, 0);
        return new ArrayList<>(powerSet);
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}

        for i, val in enumerate(nums):
            if target - val in map:
                return [map.get(target - val), i]
            map[val] = i

        return [0,0]